api/src/test/java/org/pmiops/workbench/api/ClusterControllerTest.java

```diff
@@ -223,35 +226,57 @@ private JSONObject dataUriToJson(String dataUri) {
 
   @Test
   public void testListClusters() throws Exception {
-    when(notebookService.getCluster(WORKSPACE_NS, "all-of-us")).thenReturn(testFcCluster);
+    when(notebookService.getCluster(BILLING_PROJECT_ID, "all-of-us")).thenReturn(testFcCluster);
 
-    assertThat(clusterController.listClusters().getBody().getDefaultCluster())
+    assertThat(clusterController.listClusters(BILLING_PROJECT_ID).getBody().getDefaultCluster())
         .isEqualTo(testCluster);
   }
 
   @Test
   public void testListClustersUnknownStatus() throws Exception {
-    when(notebookService.getCluster(WORKSPACE_NS, "all-of-us")).thenReturn(
+    when(notebookService.getCluster(BILLING_PROJECT_ID, "all-of-us")).thenReturn(
         testFcCluster.status(null));
 
-    assertThat(clusterController.listClusters().getBody().getDefaultCluster().getStatus())
+    assertThat(clusterController.listClusters(BILLING_PROJECT_ID).getBody().getDefaultCluster().getStatus())
         .isEqualTo(ClusterStatus.UNKNOWN);
   }
 
+  @Test(expected = FailedPreconditionException.class)
+  public void testListClustersNullBillingProject() throws Exception {
+    when(notebookService.getCluster(BILLING_PROJECT_ID, "all-of-us")).thenReturn(testFcCluster);
+
+    clusterController.listClusters(null);
+  }
+
+  @Test(expected = FailedPreconditionException.class)
+  public void testListClustersFreeTierNotReady() throws Exception {
+    when(notebookService.getCluster(BILLING_PROJECT_ID, "all-of-us")).thenReturn(testFcCluster);
+
+    User notReadyUser = new User();
+    notReadyUser.setEmail(LOGGED_IN_USER_EMAIL);
+    notReadyUser.setUserId(123L);
+    notReadyUser.setFreeTierBillingProjectName(BILLING_PROJECT_ID);
+    notReadyUser.setFreeTierBillingProjectStatusEnum(BillingProjectStatus.PENDING);
+
+    when(userProvider.get()).thenReturn(notReadyUser);
```

Comment:
i try to use the `doReturn` syntax everywhere. covers a few more cases than when - https://stackoverflow.com/a/20360269

Reply:
Interesting, thanks!

Looks like we use both patterns frequently.  This class uses `when`/`thenReturn` exclusively so in this case I'll keep it for internal consistency.